plugins/obs-ffmpeg/obs-nvenc.c

```diff
@@ -239,18 +241,57 @@ static bool nv_texture_init(struct nvenc_data *enc, struct nv_texture *nvtex)
 	return true;
 }
 
+#else
+
+struct nv_texture {
+	void *res;
+	CUdeviceptr tex;
+	size_t pitch;
+	void *mapped_res;
+};
+
+static bool nv_texture_init(struct nvenc_data *enc, struct nv_texture *nvtex)
+{
+	const bool p010 = obs_p010_tex_active();
+
+	cu->cuMemAllocPitch(&nvtex->tex, &nvtex->pitch, p010 ? enc->cx * 2 : enc->cx,
+			    enc->cy + enc->cy / 2, 16);
+
+	NV_ENC_REGISTER_RESOURCE res = {NV_ENC_REGISTER_RESOURCE_VER};
+	res.resourceType = NV_ENC_INPUT_RESOURCE_TYPE_CUDADEVICEPTR;
+	res.resourceToRegister = (void *)nvtex->tex;
+	res.width = p010;
```

Comment:
Not quite right?

Reply:
Indeed this is wrong (might've been a fix swallowed by a GNOME Builder crash), but NVENC didn't care amusingly enough. I guess when it comes to CUDA surfaces it can get that information another way.